src/API/Endpoints/Reports/Endpoint.php

```diff
@@ -142,6 +155,18 @@ public function sanitize_date( $param, $request, $key ) {
 		return $date;
 	}
 
+	public function validate_currency( $param, $request, $key ) {
+		return in_array( $param, array_keys( give_get_currencies_list() ) );
+	}
+
+	public function validate_test_mode( $param, $request, $key ) {
+		return $param === 'true' || $param === 'false';
```

Comment:
@henryholtgeerts You can use `filter_var`.

```suggestion
		return filter_var($param, FILTER_VALIDATE_BOOLEAN);
```

Reply:
Here, I'm only validating the parameter passed for test mode. The validation function only cares that a valid string was passed ('true' or 'false'), and returns true if that is the case. 

If I used filter_var here, it would return false when a request was made with 'false' as the test mode parameter, even thought 'false' is actually an expected and valid parameter.

I use boolval in the sanitization step, to cast the 'true' or 'false' string to booleans.

Reply:
See this comment: https://github.com/impress-org/givewp/pull/4728/files#r439527544